### PR TITLE
Make line search for transfinite manifold more stable

### PR DESCRIPTION
**Pull request by kronbichler**
The line search in the Newton iteration for `TransfiniteInterpolation::pull_back` in the transfinite interpolation would eagerly search too far outside the valid chart domain, leading to failures in the spherical manifold (where it is inverted). I added a test that previously failed in the `tria.refine_global(1)` call.

### CODE REVIEW

File: source/grid/manifold_lib.cc [hunk 1]
```diff
@@ -1200,6 +1200,14 @@ ::pull_back(const typename Triangulation<dim,spacedim>::cell_iterator &cell,
 
       // Line search, accept step if the residual has decreased
       double alpha = 1.;
+
+      // check if point is inside 1.2 times the unit cell to avoid
```
Comment (masterleinad) [inline source/grid/manifold_lib.cc]: Is there a particulary reason for 1.2 here?

Comment (kronbichler) [reply source/grid/manifold_lib.cc]: heuristics: I observed that allowing the points to be slightly outside the unit cell does happen during the early stages of the Newton iteration. On the other hand, being too far outside gives very strange weights (I observed weights 7.2, -4.4, -4.4, 1.2 or something like that for the case that `SphericalManifold got negative determinants on the Hessian and failed). Thus 20% outside seems like a reasonable tradeoff.

Comment (masterleinad) [reply source/grid/manifold_lib.cc]: OK.

**Pull request merged**